subprojects/analysis-kotlin-api/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/internal/SampleProvider.kt

```diff
@@ -0,0 +1,23 @@
+package org.jetbrains.dokka.analysis.kotlin.internal
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.InternalDokkaApi
+
+@InternalDokkaApi
+interface SampleProviderFactory {
+    fun build(): SampleProvider
+}
```

Comment:
My previous comment about this was resolved, but I didn't notice if any documentation was added - perhaps, I missed it?

If not, could you please add a comment somewhere, stating that SampleProvider exists because its instances are supposed to be short-lived, with a reduced scope of use, otherwise it can lead to high memory consumption / leaks? Without it, it's easy to think the factory layer is useless and can be removed.